tools/tcpaccept.py

```diff
@@ -141,10 +141,8 @@
     } else if (family == AF_INET6) {
         struct ipv6_data_t data6 = {.pid = pid, .ip = 6};
         data6.ts_us = bpf_ktime_get_ns() / 1000;
-        bpf_probe_read(&data6.saddr, sizeof(data6.saddr),
-            &newsk->__sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32);
-        bpf_probe_read(&data6.daddr, sizeof(data6.daddr),
-            &newsk->__sk_common.skc_v6_daddr.in6_u.u6_addr32);
+        __builtin_memcpy(&data6.saddr, newsk->__sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32, sizeof(data6.saddr));
```

Comment:
`newsk` is an external pointer and requires a call to `bpf_probe_read`. This will be rejected by the verifier. Same for the other copies below.

Reply:
But I had test them in 4.17 and 4.13 ,and they are works .

Reply:
Oh, right! We actually rewrite the dereferences inside the `__builtin_memcpy` so they won't be rejected by the verifier. It does add one useless `__builtin_memcpy` though:
```c
__builtin_memcpy(&data6.saddr, newsk->__sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32, sizeof(data6.saddr));
```
is rewritten to:
```c
__builtin_memcpy(&data6.saddr, ({ typeof(__be32 [4]) _val; __builtin_memset(&_val, 0, sizeof(_val)); bpf_probe_read(&_val, sizeof(_val), (u64)&newsk->__sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32); _val; }), sizeof(data6.saddr));
```
You can check the rewritten text with the `debug=4` argument.

Reply:
do we still need to change `__builtin_memcpy` to `bpf_probe_read` ?

Reply:
Yes.

Reply:
ok, have done.

Reply:
I think you committed the wrong files.

Reply:
tools/tcpaccept.py, tools/tcpconnect.py, tools/tcpretrans.py, tools/tcpdrop.py, tools/tcplife.py, tools/tcpstates.py six files.
https://github.com/iovisor/bcc/pull/1862/commits/939012dd8cb7232de4c95ade55160d4cc8c50e38 commit delete the wrong files.
you can refresh the web to see.